eng/PrepareRelease.yml

```diff
@@ -0,0 +1,43 @@
+stages:
+- stage: PrepareReleaseStage

```

Comment:
Because this is now tied into the product build, it has to produce the release assets correctly. Otherwise, you'd have to fix the publish issue and queue another build of the entire product. Consider a design where the product build produces the minimal assets and then have a separate yaml pipeline that can be run separately which takes the output of the product build. This would allow us, for example, to make minor changes to the release pipeline in the `main` branch yet publish bits from a release branch.

Reply:
This was an intentional choice, my worry is that we will want to make a breaking change to the assets between say 6.0 and 7.0 and we won't be able to publish 6.0.x because it points at the 7.0 release tool. We may end up in a world where we need to port around a few changes to spin a 6.0.x build if we have new publishing guidelines. We also have the escape hatch of making a separate pipeline that takes as parameters the source build and main publishing tool and can "patch-up" an existing build if we need to.
My goal here was to make this as self-contained as possible to the branch and move as much as possible up into the build pipeline so as little as possible can go wrong during the "release".